stdlib/public/runtime/SwiftValue.mm

```diff
@@ -300,30 +387,48 @@ - (BOOL)isEqual:(id)other {
     return NO;
   }
 
+  // `other` must also be a _SwiftValue box
   if (![other isKindOfClass:getSwiftValueClass()]) {
     return NO;
   }
 
   auto selfHeader = getSwiftValueHeader(self);
   auto otherHeader = getSwiftValueHeader(other);
 
-  auto hashableBaseType = selfHeader->getHashableBaseType();
-  if (!hashableBaseType ||
-      otherHeader->getHashableBaseType() != hashableBaseType) {
-    return NO;
+  auto hashableConformance = selfHeader->getHashableConformance();
+  if (hashableConformance) {
```

Comment:
This conditional pyramid might be more readable as `if (auto hashableConformance = ...)` etc. Maybe.

Reply:
Done.